hilti/include/rt/types/bytes.h

```diff
@@ -297,7 +337,7 @@ class Bytes : protected std::string, public hilti::rt::detail::iterator::Control
     Bytes join(const Vector<T>& parts) const {
         Bytes rval;
 
-        for ( auto i = 0; i < parts.size(); ++i ) {
+        for ( size_t i = 0; i < parts.size(); ++i ) {
```

Comment:
Just curious: Why this change?

Reply:
As written `i` is signed while `parts.size()` is unsigned. This triggers a diagnostic e.g., with clang that we previously didn't see as we only exercised this code in JIT; now this causes an compilation error from the unit test.